fvm/crypto/crypto_test.go

```diff
@@ -0,0 +1,229 @@
+package crypto_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime"
+
+	gocrypto "github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/crypto/hash"
+	"github.com/onflow/flow-go/fvm/crypto"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func TestHashWithTag(t *testing.T) {
+	t.Run("tag too long", func(t *testing.T) {
+		algorithms := []hash.HashingAlgorithm{
+			hash.SHA2_256,
+			hash.SHA2_384,
+			hash.SHA3_256,
+			hash.SHA3_384,
+		}
+
+		okTag := [32]byte{}
+		longTag := [33]byte{}
+
+		for i, algorithm := range algorithms {
+			t.Run(fmt.Sprintf("algo %d: %v", i, algorithm), func(t *testing.T) {
+				_, err := crypto.HashWithTag(algorithm, string(longTag[:]), []byte("some data"))
+				require.Error(t, err)
+			})
+
+			t.Run(fmt.Sprintf("algo %d: %v - control (tag ok)", i, algorithm), func(t *testing.T) {
+				_, err := crypto.HashWithTag(algorithm, string(okTag[:]), []byte("some data"))
+				require.NoError(t, err)
+			})
+		}
+	})
+}
+
+func TestVerifySignatureFromRuntime(t *testing.T) {
+	t.Run("verify should fail on incorrect combinations", func(t *testing.T) {
+		correctCombinations := make(map[runtime.SignatureAlgorithm]map[runtime.HashAlgorithm]struct{})
```

Comment:
I think this way would be more readable
```suggestion
				correctCombinations := map[runtime.SignatureAlgorithm]map[runtime.HashAlgorithm]struct{}{
			runtime.SignatureAlgorithmBLS_BLS12_381: map[runtime.HashAlgorithm]struct{}{
				runtime.HashAlgorithmKMAC128_BLS_BLS12_381: struct{}{},
			},
			runtime.SignatureAlgorithmECDSA_P256: map[runtime.HashAlgorithm]struct{}{
				runtime.HashAlgorithmSHA2_256: struct{}{},
				runtime.HashAlgorithmSHA3_256: struct{}{},
			},
			runtime.SignatureAlgorithmECDSA_secp256k1: map[runtime.HashAlgorithm]struct{}{
				runtime.HashAlgorithmSHA2_256: struct{}{},
				runtime.HashAlgorithmSHA3_256: struct{}{},
			},
		}
```